### ISSUE TITLE
Root path

### ISSUE THREAD

**Issue by skarllot** [emgarten/Sleet#4]
When I try to configure path to "http://nuget.example.com/" I get the following error: Object reference not set to an instance of an object

To work around this issue a have to add a folder to path as "http://nuget.example.com/feed".

**Comment by MPTG94** [comment]
Can you add your `sleet.json` file here?

**Comment by fgodoy-verion** [comment]
```json
{
  "username": "",
  "useremail": "",
  "sources": [
    {
      "name": "myLocalFeed",
      "type": "local",
      "path": "http://nuget.example.com/"
    }
  ]
}
```

**Comment by MPTG94** [comment]
@emgarten Seems like the issue is with the `CreateUri` method in the `UriUtility.cs` file,

If you run `sleet.exe` with `"path": http://nuget.example.com/` 

the command will fail and the exception will be:
`Invalid URI: The hostname could not be parsed.`

This happens on execution of the line:
`_baseUri = UriUtility.CreateUri(baseUri.AbsoluteUri.TrimEnd(new char[] { '/', '\\' }) + Path.DirectorySeparatorChar);`

in the `PhysicalFileSystem.cs` file

if you try and run with `"path": http://nuget.example.com/feed` the command will run just fine and create a new directory in the path C:\feed

I'm guessing this has to do with sleet being unable to map `http://nuget.example.com/` to a writeable filesystem path at `C:\`

**Comment by emgarten** [comment]
Thanks for debugging it @MPTG94  I'll take a look at getting this fixed.

**Comment by MPTG94** [comment]
Some more information:

@emgarten Basically the program catches an exception when executing the line:
`return new Uri(path);`

This is in the `UriUtility.cs` class in the `CreateUri(string path)` method.

An input which doesn't crash:
`http://nuget.example.com/feed\\`

An input which crashes:
`http://nuget.example.com\\`

In both cases the trailing `\\` at the end are added by the `baseUri.AbsoluteUri.TrimEnd` call

**Comment by emgarten** [comment]
I did some clean up on the Uri parsing code, the fix for this issue is available now in Sleet 1.0.1: https://github.com/emgarten/Sleet/releases/latest

Let me know if you hit any other problems with creating feeds that have different base path.s
